Lib/test/test_socket.py

```diff
@@ -1519,6 +1519,24 @@ def test_unusable_closed_socketio(self):
             self.assertRaises(ValueError, fp.writable)
             self.assertRaises(ValueError, fp.seekable)
 
+    def test_socket_close(self):
+        try:
+            sock = socket.socket()
+            sock.bind((HOST, 0))
+            socket.close(sock.fileno())
+            with self.assertRaises(OSError):
+                sock.listen(1)
+        finally:
+            try:
+                sock.close()
+            except OSError:
+                # sock.close() fails with EBADF
```

Comment:
If it always does, then how about a `assertRaises`?

Reply:
assertRaises in the finally block may mask an error in the try block.

Reply:
Normally, Python 3 would display the exception chain.